src/credentials/CHIPCert.cpp

```diff
@@ -708,22 +708,109 @@ CHIP_ERROR DetermineCertType(ChipCertificateData & cert)
 
 bool ChipDN::IsEqual(const ChipDN & other) const
 {
-    if (mAttrOID == kOID_Unknown || mAttrOID == kOID_NotSpecified || mAttrOID != other.mAttrOID)
+
+    if (mAttrOID[0] == chip::ASN1::kOID_AttributeType_ChipNodeId)
+        return other.mAttrOID[0] == chip::ASN1::kOID_AttributeType_ChipNodeId &&
+            mAttrValue[0].mChipId == other.mAttrValue[0].mChipId;
+
+    for (int i = 1; i < RDN_NUM; ++i)
     {
-        return false;
+        if (mAttrOID[i] == kOID_Unknown || mAttrOID[i] == kOID_NotSpecified)
+            continue;
+
+        int j = 1;
+        for (; j < RDN_NUM; ++j)
+        {
+            if (mAttrOID[i] == other.mAttrOID[j])
+                break;
+        }
+
+        if (j == RDN_NUM)
```

Comment:
This looks to me as if we loop to the next if this.mAttrOID is not contained in other.mAttrOID.

So if the mAttrOID are completely distinct between 2 ChipDN objects, then this method will return true. This seems unintuitive to me (if this and other are completely different, then return value is true). 

Could you explain? This method may use some comments.

Reply:
My bad, thanks for catching. Fixed